File: CPP/String/String Compression III.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp;
        int i=0;
        int n =word.size();
        while(i<n){
            int cnt =0;
            char c=word[i];
            while (i<n && c==word[i] && cnt<9){
                //if(c==word[i]){
                    cnt++;
                //}
                i++;
            }
            comp+=to_string(cnt)+c;
        }
        return comp;
    }
};

File: Python/String/String Compression III.py
class Solution:
    def compressedString(self, word: str) -> str:
        comp=[]
        n=len(word)
        i=0
        while i<n:
            cnt=0
            c=word[i]
            while i<n and c== word[i] and cnt<9:
                cnt+=1
                i+=1
            comp.append(str(cnt))
            comp.append(c)
        return "".join(comp)

        
